### DISCUSSION TITLE
v0.60.0

### DISCUSSION THREAD

**Discussion by tommoor** [outline/outline#2755] [Announcements]
> Note: This update requires database migrations, take your installation offline, backup the database, and run with yarn db:migrate or yarn db:migrate --env production-ssl-disabled

### Collaborative editing

![image](https://user-images.githubusercontent.com/380914/141026540-67401192-23f9-46f9-ac8d-9ba8fd8605dd.png)

We're excited to include the first beta of collaborative editing in this release. This feature allows multiple team members to edit the same document synchronously with all changes merged automatically. This feature also includes live cursors, presence, and selection plus the removal of the separate "Edit" mode. To opt-in visit **Settings -> Features** as an installation admin to enable. Please note that your installation must support websocket connections, and in the future this will be a requirement to continue upgrading Outline.

### Command bar

This release also includes new `cmd`+`k` functionality which you might recognize from other productivity apps. Type `cmd`+`k` anywhere to quickly move around your knowledge base! We'll be building all of the possible actions into this menu over the rest of the year. Some shortcuts of note that were added as part of this – `o` then `d` to directly jump to documents, or `o` then `c` to jump to collections!

As there was no official release in October this one is also jam-packed with lots of other smaller improvements and fixes:

### Enhancements

- New in-built emoji picker in editor, type `:` to trigger (#2611)
- Added Perl language support in editor code blocks
- Public share links now include an option to change light/dark theme (#2632)
- It is now possible to set a default role for new users in **Settings -> Security** (#2665)
- Warning when dragging doc between collections with different permissions (#2516)
- Zooming images now possible while editing
- Several new icons for collections
- It is now possible to delete previous exports in admin (#2595)
- 🇵🇱 Polish added as language option

#### Editor

- It is now possible to start lists at numbers other than 1.
- `Shift+Enter` in code block now preserves indentation (outline/rich-markdown-editor#578)
- `Shift+Tab` now moves up menus (outline/rich-markdown-editor#575)
- Fixed newline creation in folded headings (outline/rich-markdown-editor#562) 
- Fixed a race condition while uploading an image to the editor
- Fixed hanging block menu trigger in gutter (+) after folding heading (outline/rich-markdown-editor#558) 
- Keyboard shortcuts for headings and code now correctly toggle the block state
- Fixed text selection toolbar will no longer appear when text selection is empty
- Improved pointer behavior on checkboxes
- Anchor "#" links now work in publicly shared docs (#2652)
- Clicking outside of the editor will now close the toolbar
- Fixed a regression with `mailto` links (#1090)
- Improvements to gist embed display (#2697)
- Improved mobile styles (#2720)
- Fixed clash between history and table of contents shortcuts (#2733)
- Fixed UI flash when loading history sidebar
- Fixed code highlighting not updating unless current selection is within code block

#### Platform

- Emoji in document title are no longer considered when sorting (#2687)
- Slack and Email settings now more clearly communicate when they are not setup
- Much improved server logging, setting a `DEBUG` variable is no longer required (#2567)
- Server now shuts down more gracefully
- Fixed potential forward to incorrect collection url on first sign-in (#2565)
- Printing from doc menu no longer produces a blank page in Firefox (#2543)
- Metrics from worker queues are now available in statsd (#2545)
- Redis v6 on Heroku is now supported
- Starred documents can now longer be highlighted twice in sidebar (#2551) 
- OIDC `preferred_username` claim now mapped to user record (#2569)
- Context menus can no longer extend out of window bounds (#2492)
- Creating an API token no longer reloads the app (#2604)
- Public share information on doc not displaying correctly after loading a shared child document
- Updated community translations

<hr /><em>This discussion was created from the release <a href='https://github.com/outline/outline/releases/tag/v0.60.0'>v0.60.0</a>.</em>

**Comment by raldred** [comment]
This is a great update.
Getting 'WebSocket connection to 'wss://domain.com/collaboration/document.0ed42643-7a7e-4da9-9681-87a1ea370142' failed: ' there's very little else in the console to aid debugging.
Hosted on Heroku

**Reply by Snuupy** [reply]
I tried https://github.com/outline/outline/discussions/2222#discussioncomment-864452 but it does not work. Not sure how to get websockets working with nginx.

I also tried https://github.com/chsasank/outline-wiki-docker-compose/issues/1#issue-658006313 but that didn't work either.

```
WebSocket connection to 'wss://outline.domain.com/realtime/?EIO=3&transport=websocket' failed: 
```

**Reply by Snuupy** [reply]
Got it working! In my setup, I have 2 reverse proxies: the first to handle SSL termination and the 2nd to handle outline. Both need the following lines added:

```
    proxy_set_header Host $host;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
```

My mistake was that I only added the above to the first (SSL termination) proxy, but not the 2nd one. Adding it to both got me the collaboration feature working! :)
